contracts/lpp/src/lpp.rs

```diff
@@ -796,8 +858,46 @@ mod test {
         assert_eq!(withdraw, Coin::new(1110));
     }
 
-    fn coin_cw(amount: u128) -> CwCoin {
-        coin_legacy::to_cosmwasm::<TheCurrency>(amount.into())
```

Comment:
this change triggers a lot of updated lines of code ... what's your motivation? If it was to accept Amount and Coin<TheCurrency> then define the arg as `Into<Coin<Currency>>`